fuse_optimizers/src/fixed_lag_smoother.cpp

```diff
@@ -138,17 +138,19 @@ void FixedLagSmoother::preprocessMarginalization(const fuse_core::Transaction& n
   timestamp_tracking_.addNewTransaction(new_transaction);
 }
 
-std::vector<fuse_core::UUID> FixedLagSmoother::computeVariablesToMarginalize()
+ros::Time FixedLagSmoother::computeLagExpirationTime() const
 {
-  auto current_stamp = timestamp_tracking_.currentStamp();
-  auto lag_stamp = ros::Time(0, 0);
-  if (current_stamp > ros::Time(0, 0) + params_.lag_duration)
-  {
-    lag_stamp = current_stamp - params_.lag_duration;
-  }
-  auto old_variables = std::vector<fuse_core::UUID>();
-  timestamp_tracking_.query(lag_stamp, std::back_inserter(old_variables));
-  return old_variables;
+  // Find the most recent variable timestamp
+  auto now = timestamp_tracking_.currentStamp();
+  // Then carefully subtract the lag duration. ROS Time objects do not handle negative values.
+  return (ros::Time(0, 0) + params_.lag_duration > now) ? ros::Time(0, 0) : now - params_.lag_duration;
```

Comment:
```suggestion
  return now > ros::Time(params_.lag_duration.toSec()) ? now - params_.lag_duration : ros::Time(0, 0);
```
A slightly different implementation for this, that doesn't subtract `params_.lag_duration` from `now` if they're the same, since the result would also be `ros::Time(0, 0)`.

Reply:
My intention was to handle that case correctly, but I see I have my equality backwards. How about:
```suggestion
  return (ros::Time(0, 0) + params_.lag_duration < now) ? now - params_.lag_duration : ros::Time(0, 0);
```
I hate bouncing between floating point and back if I don't have to.

Reply:
That sounds good :+1: